### PR TITLE
Add unit tests that fail for nested groups (the bugs are already fixed)

### PR DESCRIPTION
**Pull request by adamkaplan**
## Description
Add unit tests to prove that nested groups are configured properly.

## Motivation and Context
These test prove fixes from #13 and #14.

## How Has This Been Tested?
Local unit tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## License
I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: Example-Swift/Tests/FeatureRegistrySpec.swift [hunk 1]
```diff
@@ -42,6 +42,34 @@ class FeatureRegistrySpec: QuickSpec {
...
+                let featureNames = registry.features.map { return $0.label }
+
+                expect(featureNames).to(equal([ "group1" ]))
+
+                expect(registry.group1.feature1_1.enabled).to(beFalse())
+                expect(registry.group1.feature1_1.key).to(equal("KEY_TEST"))
+                expect(registry.group1.feature1_1.override).to(equal(OverrideState.featureDefault))
+
+                expect(registry.group1.group1_1.feature1_1_1.enabled).to(beFalse())
+                expect(registry.group1.group1_1.feature1_1_1.key).to(equal("feature1_1_1"))
```
Comment (adamkaplan) [inline Example-Swift/Tests/FeatureRegistrySpec.swift:69]: This test fails before PR #14

File: Example-Swift/Tests/FeatureRegistrySpec.swift [hunk 2]
```diff
@@ -117,28 +145,91 @@ class FeatureRegistrySpec: QuickSpec {
...
+                public let feature2 = Feature()
+                public let feature3 = Feature()
+                public let feature4 = Feature()
+                public let groupFeature1 = TestFeatureGroup()
             }
+
+            let registry = TestRegistry(withFeatureStore: nil)
+            registry.feature3.override = .enabled
+            let enabledFeatureNames = TestRegistry.enabledFeatures(in: registry)
+            expect(enabledFeatureNames).to(equal([ "feature1",
```
Comment (adamkaplan) [inline Example-Swift/Tests/FeatureRegistrySpec.swift:229]: This test fails before the final commits in PR #13

File: Example-Swift/Tests/FeatureRegistrySpec.swift [hunk 3]
```diff
@@ -117,28 +145,91 @@ class FeatureRegistrySpec: QuickSpec {
...
+                let registry = TestRegistry(withFeatureStore: nil)
+                let featureNames = registry.features.map { return $0.label }
+                expect(featureNames).to(equal([ "feature1",
+                                                "group1" ]))
+            }
+
+            it("Extracts features, nested groups") {
+                func reduce(items: [LabeledItem]) -> [String] {
+                    return items.flatMap { item -> [String] in
+                        switch (item) {
```
Comment (Eli-Liebman) [inline Example-Swift/Tests/FeatureRegistrySpec.swift:163]: I forgot we could switch over the items! I like this better than the `if let` pattern I used, I plan on revisiting it anyway.

**Review by Eli-Liebman — APPROVED** [review]
Looks good 👍

**Pull request merged**